### PR TITLE
store/tikv: handle the error of "mismatch cluster id"

### PR DESCRIPTION
**Pull request by zimulala**
## What have you changed? (mandatory)

We will exit the current TiDB when we encounter the error of "mismatch cluster id, need 6577643568868929567 but got 6577623361857143480".
The TiDB will meet the error of “mismatch cluster id” when the cluster ID of the TiDB‘s PD client does not match the PD.

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
In local

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Does this PR need to be added to the release notes? (mandatory)

```
release note:
Exit the current TiDB when it encounters the error of "mismatch cluster id".

```

### CODE REVIEW

File: store/tikv/backoff.go [hunk 1]
```diff
@@ -233,6 +234,9 @@ func (b *Backoffer) Backoff(typ backoffType, err error) error {
 			}
 		}
 		log.Warn(errMsg)
+		if strings.Contains(errMsg, ErrMismatchClusterID.Error()) {
```
Comment (shenli) [inline store/tikv/backoff.go]: Can we define a string constant for "mismatch cluster id"?

Comment (shenli) [reply store/tikv/backoff.go]: Is it better to check and handle this in the pd-client? @disksing What do you think about it?

Comment (disksing) [reply store/tikv/backoff.go]: I think we can check it at the beginning of this function.

Comment (shenli) [reply store/tikv/backoff.go]: Good idea.

File: store/tikv/error.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ import (
 var (
 	// ErrBodyMissing response body is missing error
 	ErrBodyMissing = errors.New("response body is missing")
+	// ErrMismatchClusterID returns the error that the cluster ID of the PD client does not match the PD.
+	ErrMismatchClusterID = errors.New("mismatch cluster id")
```
Comment (coocood) [inline store/tikv/error.go]: We don't need to define an error if we never throw it.
I think just make the string constant and check the string will do.

Comment (shenli) [reply store/tikv/error.go]: Agree

**Comment by zimulala** [comment]
PTAL @disksing @shenli @coocood

**Comment by disksing** [comment]
LGTM

**Pull request merged**